docs/ecs/proposal.md

```diff
@@ -0,0 +1,23 @@
+This is an enhancement to KRUN to optionally trigger KM with a docker label/OCI annotation.
+
+## Background
+
+KRUN is derived from CRUN and Kontain-specific logic is triggered based on `argv[0]` (the name of the executable). If name is 'krun` then kontain specific logic is run to mount the virtualization device (kvm or kkm) in the container and to start the entrypoint under the control of km. Otherwise, the container is started as normal.
+
+This is aligned with standard OCI based orchestration systems like kubernetes as well as with container systems that support multiple user settable runtimes like docker and podman.
+
+This doesn't work so well with AWS ECS however.
+
+## The Problem with AWS ECS
+
+ECS does not expose multiple user settable runtime. It's model is each node has a single type of runtime and all containers that run on that node run under that runtime, including a special Amazon provided `ecs-agent` container.
+
+ECS deals in docker images. Docker images can contain labels of the form `<key>=<value>` and an ECS container inherits the labels the its parent image.
+
+## Proposal
+
+Kontain payload images are identified with the label `app.kontain.version=1`. The label is introduced in `payloads/busybox/runenv.dockerfile` and `payloads/dynamic-base-scratch/runenv.dockerfile` so is therefore inherited by the rest of the of the kontain payload containers.
+
+A new name of `krun`, `krun-ecs` will be recognized. When started as `krun-ecs` the decision to use KM or
```

Comment:
The name of the runtime itself is irrelevant. The name of the executable file is what's important.

Reply:
crun and krun are the same executable for our kontainers.
currently when crun starts up it looks at its argv[0] to see what name it was started with.
if started with krun it will force km to run the entry point of the container.
if started with crun the entry point of the container is started directly.
this proposal causes crun or krun to look for the mentioned annotation from the container being started.
if the annotation is found and has the correct value, then crun starts km which starts the container entry point.

Reply:
Disregard I guess I'm wrong.